CHANGELOG.md

```diff
@@ -4,6 +4,151 @@ The format is based on [Keep a Changelog].
+		.register_subscription("sub", "s", "unsub", |_, pending, _| async move {
+			pending.accept().await?;
+
+			let stream = stream_of_integers();
+
+			loop {
+				tokio::select! {
+					_ = sink.closed() => break Ok(()),
+					maybe_item = stream.next() => {
+						let Some(item) = maybe_item else {
+							break Ok(())
+						};
+						let msg = SubscriptionMessage::from_json(&item)?;
+						if let Err(e) = sink.send_timeout(msg) {
+							match e {
+								// The subscription is closed.
+								Err(TrySendError::Closed(_)) => break Ok(()),
+								// Channel is full, just drop the subscription
+								Err(TrySendError::Full(_)) => break Ok(()),
+							}
+						}
+					}
+				}
+			}
+		})
+		.unwrap();
+```
+
+### Method call return type is more flexible
+
+This release also introduces a trait called `IntoResponse` which is makes it possible to return custom types and/or error
+types instead of enforcing everything to return `Result<T, jsonrpsee::core::Error>`
+
+This affects the APIs `RpcModule::register_method`, `RpcModule::register_async_method` and `RpcModule::register_blocking_method`
+and when these are used in the proc macro API are affected by this change.
+Be aware that [the client APIs don't support this yet](https://github.com/paritytech/jsonrpsee/issues/1067)
+
+The `IntoResponse` trait is already implemented for `Result<T, jsonrpsee::core::Error>` and for the primitive types
+
+Before it was possible to do:
+
+```rust
+	// This would return Result<&str, jsonrpsee::core::Error>
+	module.register_method("say_hello", |_, _| Ok("lo"))?;
+```
+
+After this release it possible to do:
+
+```rust
+	// Note, this method call is infallible and you might not want to return Result.
+	module.register_method("say_hello", |_, _| "lo")?;
+```
+
+### Subscription API is changed.
+
+jsonrpsee now spawns the subscriptions via `tokio::spawn` and it's sufficient to provide an async block in `register_subscription`
+
+Further, the subscription API had an explicit close API for closing subscriptions which was hard to understand and
+to get right. This has been removed and everything is handled by the return value/type of the async block instead.
+
+Example:
+
+```rust
+	module
+		.register_subscription::<RpcResult<(), _, _>::("sub", "s", "unsub", |_, pending, _| async move {
+			// This just answers the RPC call and if this fails => no close notification is sent out.
+			pending.accept().await?;
+			// This is sent out as a `close notification/message`.
+			Err(anyhow::anyhow!("The subscription failed"))?;
+		})
+		.unwrap();
+```
+
+The return value in example above needs to implement `IntoSubscriptionCloseResponse` and in this case it's `Result<(), E>` which
+implies that any value after `pending.accept().await?`.
+
+Because `Result<(), E>` is used here then close notification will sent out as error notification and it's also possible
+to ignore the return value by using `()` or implement `IntoSubscriptionCloseResponse` for some other behaviour.
```

Comment:
```suggestion
Because `Result<(), E>` is used here then close notification will sent out as error notification and in this case
the return value is ignored by using `()`. You can implement `IntoSubscriptionCloseResponse` for some other behaviour.
```

Reply:
Something like that; just grammar nits :)